Tests/BackendIntegrationTests/StoreKitObserverModeIntegrationTests.swift

```diff
@@ -81,6 +81,33 @@ class StoreKit2ObserverModeIntegrationTests: StoreKit1ObserverModeIntegrationTes
 
 }
 
+class StoreKit2ObserverModeDisabledIntegrationTests: StoreKit1ObserverModeIntegrationTests {
```

Comment:
This is going to inherit and run every single test in the super class.

Reply:
Ugh. This can be tested as unit in `StoreKitTests` (because we need a `Transaction` object) but we don't have any test that instantiates a Purchases object. Is it worth it creating one?

Reply:
> but we don't have any test that instantiates a Purchases object. Is it worth it creating one?

Yeah I think it would be useful, but also mock all the dependencies. So I think we can extract the `Purchases` setup from `BasePurchasesTest` for that.

Reply:
true! I moved it to `StoreKit2IntegrationTests` and created `StoreKit2NotEnabledObserverModeIntegrationTests` for the sk2 not enabled one.